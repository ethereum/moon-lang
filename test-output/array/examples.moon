Array = zb2rhXsv3emLb5JadcgLjdGnmU6jQD9BorHPwADKLQ7JHkmEb
a = [1 2 3 4]
b = [5 6 7 8]
{
  get: (Array "get" a "0")
  concat: (Array "concat" a b)
  dot: (Array "dot" [3 4 0] [3 4 0])
  flatMap: (Array "flatMap" (Array "reverse") [[1 2] [3 4] [5 6]])
  foldr: (Array "foldr" (add) 0 a)
  join: (Array "join" "-" ["foo" "bar" "tic" "tac"])
  map: (Array "map" (add 1) a)
  range: (Array "range" 0 6)
  reverse: (Array "reverse" a)
  sum: (Array "sum" a)
  toList: (Array "toList" a)
  zipWith: (Array "zipWith" (add) a b)
  crossWith: (Array "crossWith" (add) 0 [[1 2 3] [1 1 1] [1 1 1]])
}
