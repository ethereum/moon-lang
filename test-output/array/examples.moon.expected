{
  get: (get [1 2 3 4] (nts "0"))
  concat: (gen
    a => b => 
      (a 
        "length"
        (sub (add (len [1 2 3 4]) (len [5 6 7 8])) 0)
        (for 0 (add (len [1 2 3 4]) (len [5 6 7 8])) b c => 
          (a 
            (nts
              (sub
                (add 0 (add (len [1 2 3 4]) (len [5 6 7 8])))
                (add c 1)
              )
            )
            (if
              (ltn
                (sub
                  (add 0 (add (len [1 2 3 4]) (len [5 6 7 8])))
                  (add c 1)
                )
                (len [1 2 3 4])
              )
              (get
                [1 2 3 4]
                (nts
                  (sub
                    (add 0 (add (len [1 2 3 4]) (len [5 6 7 8])))
                    (add c 1)
                  )
                )
              )
              (get
                [5 6 7 8]
                (nts
                  (sub
                    (sub
                      (add 0 (add (len [1 2 3 4]) (len [5 6 7 8])))
                      (add c 1)
                    )
                    (len [1 2 3 4])
                  )
                )
              )
            )
          )
        )
      )
  )
  dot: 25
  flatMap: [2 1 4 3 6 5]
  foldr: 10
  join: "foo-bar-tic-tac"
  map: [2 3 4 5]
  range: [0 1 2 3 4 5]
  reverse: [4 3 2 1]
  sum: 10
  toList: a => b => (a 1 (a 2 (a 3 (a 4 b))))
  zipWith: [6 8 10 12]
  crossWith: [3 4 5]
}
